sinabs/backend/dynapcnn/utils.py

```diff
@@ -204,297 +182,56 @@ def merge_conv_bn(conv, bn):
-        layers, layer_idx_next
-    )
-    # Increment layer index to after the pooling layers
-    layer_idx_next = i_next
-
-    # Compose DynapcnnLayer
-    dynapcnn_layer = DynapcnnLayer(
-        conv=lyr_conv,
-        spk=lyr_spk,
-        pool=lyr_pool,
-        in_shape=in_shape,
-        discretize=discretize,
-        rescale_weights=rescale_factor,
-    )
-
-    return dynapcnn_layer, layer_idx_next, rescale_factor_after_pooling
-
-
-def build_from_list(
-    layers: List[nn.Module],
-    in_shape,
-    discretize=True,
-    dvs_input=False,
-) -> nn.Sequential:
-    """Build a sequential model of DVSLayer and DynapcnnLayer(s) given a list of layers comprising
-    a spiking CNN.
+def merge_conv_bn(conv: nn.Conv2d, bn: nn.BatchNorm2d) -> nn.Conv2d:
+    """Merge a convolutional layer with subsequent batch normalization.
 
     Parameters
     ----------
-
-        layers: sequence of layer objects
-        in_shape: tuple of integers
-            Shape of the input to the first layer in `layers`. Convention:
-            (channels, height, width)
-        discretize: bool
-            Discretize weights and thresholds if True
-        dvs_input: bool
-            Whether model should receive DVS input. If `True`, the returned model
-            will begin with a DVSLayer with `disable_pixel_array` set to False.
-            Otherwise, the model starts with a DVSLayer only if the first element
-            in `layers` is a pooling, cropping or flipping layer.
+        conv: torch.nn.Conv2d
+            Convolutional layer
+        bn: torch.nn.Batchnorm2d
+            Batch normalization
 
     Returns
     -------
-        nn.Sequential
+        torch.nn.Conv2d: Convolutional layer including batch normalization
     """
-    compatible_layers = []
-    lyr_indx_next = 0
-    # Find and populate dvs layer (NOTE: We are ignoring the channel information here and could lead to problems)
-    dvs_layer, lyr_indx_next, rescale_factor = construct_dvs_layer(
-        layers, input_shape=in_shape, idx_start=lyr_indx_next, dvs_input=dvs_input
-    )
-    if dvs_layer is not None:
-        compatible_layers.append(dvs_layer)
-        in_shape = dvs_layer.get_output_shape()
-    # Find and populate dynapcnn layers
-    while lyr_indx_next < len(layers):
-        if isinstance(layers[lyr_indx_next], DEFAULT_IGNORED_LAYER_TYPES):
-            # - Ignore identity, dropout and flatten layers
-            lyr_indx_next += 1
-            continue
-        dynapcnn_layer, lyr_indx_next, rescale_factor = construct_next_dynapcnn_layer(
-            layers,
-            lyr_indx_next,
-            in_shape=in_shape,
-            discretize=discretize,
-            rescale_factor=rescale_factor,
-        )
-        in_shape = dynapcnn_layer.get_output_shape()
-        compatible_layers.append(dynapcnn_layer)
-
-    return nn.Sequential(*compatible_layers)
+    return merge_bn(conv, bn)
```

Comment:
`merge_conv_bn` is for backward compatibility. We can also deprecate it.

Reply:
Conclusion: Keep `merge_conv_bn` with deprecation warning. Inside our code use `merge_bn`. Remove `merge_linear_bn`, since it has not existed before and adds no new functionality.